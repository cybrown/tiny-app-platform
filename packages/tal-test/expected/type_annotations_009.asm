main()
  entry:
    DeclareLocal      name: a, mutable: true, hasInitialValue: false
    Pop               inBlock: false
    Literal           3
    DeclareLocal      name: b, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal           3
    DeclareLocal      name: c, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal           "a"
    Local             a
    Literal           "b"
    Local             b
    Literal           "c"
    Local             c
    Literal           3
    MakeRecord
