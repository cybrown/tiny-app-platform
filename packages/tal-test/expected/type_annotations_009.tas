var a: number
var b: number = 3
var c = 3

{a: a, b: b, c: c}
