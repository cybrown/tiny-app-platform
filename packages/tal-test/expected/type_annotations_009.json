[
  {
    "kind": "DeclareLocal",
    "mutable": true,
    "name": "a",
    "type": {
      "kind": "named",
      "name": "number"
    }
  },
  {
    "kind": "DeclareLocal",
    "mutable": true,
    "name": "b",
    "type": {
      "kind": "named",
      "name": "number"
    },
    "value": 3
  },
  {
    "kind": "DeclareLocal",
    "mutable": true,
    "name": "c",
    "value": 3
  },
  {
    "kind": "Record",
    "entries": [
      {
        "kind": "RecordEntry",
        "key": "a",
        "value": {
          "kind": "Local",
          "name": "a"
        }
      },
      {
        "kind": "RecordEntry",
        "key": "b",
        "value": {
          "kind": "Local",
          "name": "b"
        }
      },
      {
        "kind": "RecordEntry",
        "key": "c",
        "value": {
          "kind": "Local",
          "name": "c"
        }
      }
    ]
  }
]